### PR TITLE
fix(tailwind): collapse empty-fallback var() refs in inline styles

### LINKED ISSUE #2898: `font-variant-numeric` not rendering correctly
### What versions are you using? (if relevant)

@react-email/components@1.0.6, react-email@5.2.5, etc.

### Describe the Bug

When using the tailwind classes for `font-variant-numeric`, for example `tabular-nums`, the output is not formatted correctly

```
 <p className="tabular-nums">
    1234
</p>
```

Becomes

```                            
<p
  style="font-variant-numeric:var(--tw-ordinal,) var(--tw-slashed-zero,) var(--tw-numeric-figure,) tabular-nums var(--tw-numeric-fraction,)">
  1234
</p>
```

### What is affected (leave empty if unsure)

_No response_

### Link to the code that reproduces this issue

tabular-nums.tsx
```
import { Html, Body, Tailwind, pixelBasedPreset } from "@react-email/components";
import * as React from "react";

export default function SocietEmail() {
    return (
        //@ts-ignore
        <Tailwind
            config={{
                presets: [pixelBasedPreset],
            }}
        >
            <Html>
                <Body>
                    <p className="tabular-nums">
                    1234
                    </p>
                </Body>
            </Html>
        </Tailwind>
    );
}
```

### To Reproduce

Use the provided template provided above

### Expected Behavior

Should probably convert to the following, not certain on this though
```
font-variant-numeric:tabular-nums
```

### What's your node version? (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by mvanhorn**
Closes #2898

## Repro

With the `Tailwind` component and a `tabular-nums` className:

```tsx
<Tailwind><p className="tabular-nums">1234</p></Tailwind>
```

Rendered output (before this PR):

```html
<p style="font-variant-numeric:var(--tw-ordinal,) var(--tw-slashed-zero,) var(--tw-numeric-figure,) tabular-nums var(--tw-numeric-fraction,)">1234</p>
```

No email client reliably evaluates CSS custom properties, so most inboxes drop the entire `font-variant-numeric` declaration and the user never sees tabular figures.

## Root cause

Tailwind v4 compiles every `font-variant-numeric` utility into a variant-stacking chain. Each optional slot is an unresolved `var(--tw-<name>,)` with an empty fallback, so when a variant isn't in use it collapses to nothing per the CSS Custom Properties spec: "If the declared property does not have a value, substitute the fallback value" -- empty fallback = empty string.

`makeInlineStylesFor` already tries to substitute variables, but only when the custom property has an `initialValue` registered on `:root`. Tailwind v4 deliberately leaves these variant vars undefined, so the walker leaves the `var()` calls in place, and the `generate()` call writes them to the inline style object as-is.

## Fix

After `generate(declaration.value)`, strip any leftover `var(--name,)` calls with empty fallbacks and collapse the leftover whitespace before committing the style to the output map. Diff: +7 lines of logic + rationale comment in `make-inline-styles-for.ts`.

Scoped regex: only matches `var(` + `--...` + `,` + `)` (empty fallback). Non-empty fallbacks and `var(--foo)` without a fallback are untouched, so this doesn't change existing resolution paths.

## Rendered output (after)

```html
<p style="font-variant-numeric:tabular-nums">1234</p>
```

## Tests

- New `strips Tailwind v4 variant-stacking var() refs with empty fallbacks` test in `make-inline-styles-for.spec.ts` covering the exact output Tailwind v4 emits for `tabular-nums`.
- Re-snapshotted `does basic local variable resolution`: the previously retained leading space (`" #3490dc"`) was an accidental artifact of `generate()` - the new whitespace collapse makes the output `"#3490dc"` without the leading space. Functionally identical for React's inline style map.
- All 87 `packages/react-email/src/components/tailwind/` tests pass locally.

## Scope

- One function touched. Does not affect `resolveAllCssVariables` (which already has its own fallback handling at `resolve-all-css-variables.ts:192`).
- Does not fix the same class of bug for other Tailwind multi-var properties (`transform`, `filter`, `backdrop-filter`) - those still have the same pattern but only `font-variant-numeric` is in the issue. Follow-up welcome; regex is class-agnostic so it may Just Work for those if exercised.

This contribution was developed with AI assistance (Claude Code).

---
## Summary by cubic
Removes empty-fallback `var()` refs from Tailwind v4 variant-stacking in inline styles (e.g., `tabular-nums`) so `font-variant-numeric` works in emails. Scoped to `--tw-*` vars, trims whitespace, and handles nested `var()` chains.

- **Bug Fixes**
  - Uses a `css-tree` AST post-order walk to delete `var(--tw-*,)` (including nested cases) with a targeted Raw-fallback scrubber; preserves `!important`, non-empty fallbacks, and non-`--tw-*` vars.
  - Adds tests for Tailwind v4 `tabular-nums`, nested `var(--tw-outer, var(--tw-inner,))`, and preserving user-authored `var(--my-color,)`/`var(--brand,)`; updates one whitespace snapshot.

<sup>Written for commit 62adf48ed02d9425a84b5625bc3fcb4d62abbdcc. Summary will update on new commits. <a href="https://cubic.dev/pr/resend/react-email/pull/3359?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: packages/react-email/src/components/tailwind/utils/css/make-inline-styles-for.ts [hunk 1]
```diff
@@ -59,9 +59,23 @@ export function makeInlineStylesFor(
...
+        // The walker above replaces var() calls with an initialValue when one is defined,
+        // but Tailwind deliberately leaves these variant vars undefined until used, so they
+        // stay in the output here and produce unresolvable custom properties in email HTML
+        // (no email client supports CSS custom properties reliably). Per the CSS spec
+        // (https://www.w3.org/TR/css-variables-1/#using-variables) an empty fallback means
+        // "use empty string if the variable is undefined", which is exactly what we want at
+        // inline-style time.
+        const rawValue = generate(declaration.value);
+        const cleanedValue = rawValue
+          .replace(/var\(\s*--[\w-]+\s*,\s*\)/g, ' ')
```
Comment (gabrielmfern) [inline packages/react-email/src/components/tailwind/utils/css/make-inline-styles-for.ts]: can this be removed using css tree? I believe they allow us to parse functions like this

Comment (gabrielmfern) [inline packages/react-email/src/components/tailwind/utils/css/make-inline-styles-for.ts]: the main problem with this is that the user might want to do CSS variables themselves, and they might want to use them in tailwind utilities, so this would remove those.

**Comment by mvanhorn** [comment]
Scoped the regex to `--tw-*` specifically in d73e8156 so user-authored empty-fallback vars pass through untouched. Added a spec that asserts `var(--my-color,)` and `var(--brand,)` are preserved while `var(--tw-custom,)` collapses.

On css-tree: happy to switch to an AST-based walk if you'd rather not carry the regex. The regex felt like the minimal shape for this specific Tailwind v4 idiom, but I don't have a strong preference.

**Review by gabrielmfern — APPROVED** [review]

**Comment by gabrielmfern** [comment]
> On css-tree: happy to switch to an AST-based walk if you'd rather not carry the regex. The regex felt like the minimal shape for this specific Tailwind v4 idiom, but I don't have a strong preference.

let's go for csstree here

**Review by gabrielmfern — CHANGES_REQUESTED** [review]
can you replace the regex approach with a csstree based one?

**Comment by mvanhorn** [comment]
Done in 0c280fb @gabrielmfern. Walking `declaration.value` with `visit: 'Function'` before serialization, removing `var(--tw-*, )` calls whose first Identifier starts with `--tw-` and whose argument list ends with an empty fallback (comma Operator followed by no value-bearing nodes). The two existing tests pass with the original snapshots — the regex implementation already produced `var(--brand,)` for the user-authored case so no test changes were needed. Verified with `vitest run make-inline-styles-for.spec.ts` (4/4) and `biome check` clean.

**Comment by mvanhorn** [comment]
@gabrielmfern Followed up on the cubic-dev-ai P2 in 62adf48e.

The headline change is `enter` -> `leave` in the function visitor, so children are processed before the parent's empty-fallback check runs. For the `var(--tw-outer, var(--tw-inner,))` shape that means the inner collapses first, then the outer's `hasFallbackContent` correctly reads as empty.

One honest disclosure: I also added a small Raw-pattern scrubber (lines 105-117 of `make-inline-styles-for.ts`) for the case where css-tree parses the var() fallback contents as a `Raw` node rather than nested `Function` nodes. In that shape, post-order traversal can't reach the inner `var()` because the walker never visits it. The scrubber is scoped to `--tw-*` identifiers via `/var\(--tw-[^,()]+,\s*\)/g` and only fires when `child.type === "Raw"`, so user-authored `var()` refs in the fallback (the case you flagged earlier) still pass through. If you'd prefer to keep this strictly csstree-only and accept the residual unresolved `var()` in that Raw-shape edge case, I can drop those lines and the regression test still passes for the structurally-parsed case.

Verification: `pnpm --filter react-email test make-inline-styles-for` - 5 passed (4 existing + 1 new regression test for the nested case).

**Review by gabrielmfern — APPROVED** [review]
